File: 10.Math数学/leetcode_3871_countCommas.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    long long countCommas(long long n) {
        long long res = 0;
        long long cur = 1000;
        while (cur <= n){
            res += n-cur+1;
            cur *= 1000;
        }
        return res;
    }
};

int main(){
    Solution s;
    long long n = 1002;
    cout << s.countCommas(n) << endl;
}

File: 10.Math数学/leetcode_3871_countCommas.py
class Solution:
    def countCommas(self, n: int) -> int:
        res = 0
        cur = 1000
        while cur <= n:
            res += n-cur+1
            cur *= 1000
        return res

if __name__ == "__main__":
    s = Solution()
    n = 1002
    print(s.countCommas(n))
